### PR TITLE
Adds workflow to create documentation issues

### PR DESCRIPTION
**Pull request by downsrob**
Signed-off-by: Clay Downs <downsrob@amazon.com>

### Description
Adds a workflow to open issues in the documentation-website repo when the "needs-documentation" label is added.
 
### Issues Resolved
https://github.com/opensearch-project/job-scheduler/issues/174
 
### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [X] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/documentation.md [hunk 1]
```diff
@@ -0,0 +1,15 @@
+**Is your feature request related to a problem?**
```
Comment (VachaShah) [inline .github/ISSUE_TEMPLATE/documentation.md:1]: The issue in `documentation-website` will be created with this content. Maybe use this template https://github.com/opensearch-project/OpenSearch/pull/2929/files#diff-cafe2730d029a63a9cc0fa0d9f853fca6811942e8fea2ba3b13ec08ae038dfdb so the static content is relevant?

Comment (downsrob) [reply .github/ISSUE_TEMPLATE/documentation.md:1]: Sure, I just used the current issue template for the documentation-website, but I'll make it more relevant

Comment (downsrob) [reply .github/ISSUE_TEMPLATE/documentation.md:1]: Sure, I just used the current issue template for the documentation-website, but I'll make it more relevant

**Comment by dbbaughe** [comment]
This is great!

As an obvious note it seems like all repositories would benefit from this. So as a follow up let's perhaps start looking into how we can refactor all the common reusable workflows into a single place:
- https://github.blog/2022-02-10-using-reusable-workflows-github-actions/
- https://docs.github.com/en/actions/using-workflows/sharing-workflows-secrets-and-runners-with-your-organization

This way we can have the actual logic you've written in one place and everyone can use it and if any updates are required we only need to do it at the source. This probably applies to a lot of the GitHub workflows that have been copy pasted all over our repositories.

**Review by dbbaughe — APPROVED** [review]

**Review by bowenlan-amzn — APPROVED** [review]

**Review by VachaShah — APPROVED** [review]

**Pull request merged**